### PR TITLE
Join births by year

### PR DESCRIPTION
**Pull request by seb231**
Output all historic and projected years for births projection

### CODE REVIEW

File: src/witan/models/dem/ccm/core/projection_loop.clj [hunk 1]
```diff
@@ -140,10 +140,28 @@
   [{:keys [latest-year-popn population]} _]
   {:population (ds/join-rows population latest-year-popn)})
 
+(defworkflowfn join-births-year
```
Comment (acron0) [inline src/witan/models/dem/ccm/core/projection_loop.clj]: Go on, lets have a `-1-0-0` on the end of that fn name ;)

File: src/witan/models/dem/ccm/models.clj [hunk 2]
```diff
@@ -64,8 +64,9 @@
     [:age-on                     :add-births]
     [:add-births                 :remove-deaths]
     [:remove-deaths              :apply-migration]
-    [:apply-migration            :join-popn-latest-year]
-    [:join-popn-latest-year      [:finish-looping? :out :select-starting-popn]]
+    [:apply-migration            :join-births-year]
+    [:join-births-year      :join-popn-latest-year]
+    [:join-popn-latest-year [:finish-looping? :out :select-starting-popn]]
```
Comment (acron0) [inline src/witan/models/dem/ccm/models.clj]: @sunnytownsend this all good?

File: src/witan/models/dem/ccm/core/projection_loop.clj [hunk 3]
```diff
@@ -140,10 +140,28 @@
...
+  "Takes in a dataset of births for previous years and a dataset of
+   projected births for the next year of projection.
+   Returns a datasets that appends the second dataset to the first one."
+  {:witan/name :ccm-core/join-years-births
+   :witan/version "1.0.0"
+   :witan/input-schema {:births BirthsBySexSchema
+                        :historic-births BirthsSchema
+                        :loop-year (s/constrained s/Int m-utils/year?)}
+   :witan/output-schema {:historic-births BirthsSchema}
+   :witan/exported? true}
```
Comment (acron0) [inline src/witan/models/dem/ccm/core/projection_loop.clj]: Remove `:witan/exported?` - it's legacy and ignored

**Review by acron0 — CHANGES_REQUESTED** [review]
These changes to the workflow will be undone next CC when we redo it all, minus accretion. This change heavily depends on accretion so will be directly affected.

See comments for changes

File: src/witan/models/dem/ccm/core/projection_loop.clj [hunk 4]
```diff
@@ -127,34 +127,26 @@
                             (ds/rename-columns {:popn-migrated :popn}))]
     {:latest-year-popn popn-w-migrants}))
 
-(defworkflowfn join-popn-latest-year
+(defworkflowfn join-by-year
```
Comment (acron0) [inline src/witan/models/dem/ccm/core/projection_loop.clj]: Add version please

File: src/witan/models/dem/ccm/core/projection_loop.clj [hunk 5]
```diff
@@ -127,34 +127,26 @@
...
-   :witan/input-schema {:births BirthsBySexSchema
+                        :population PopulationSchema
+                        :births BirthsBySexSchema
                         :historic-births BirthsSchema
                         :loop-year (s/constrained s/Int m-utils/year?)}
-   :witan/output-schema {:historic-births BirthsSchema}}
-  [{:keys [births historic-births loop-year]} _]
+   :witan/output-schema {:population PopulationSchema
+                         :historic-births BirthsSchema}
+   :witan/exported? true}
```
Comment (acron0) [inline src/witan/models/dem/ccm/core/projection_loop.clj]: Remove legacy key

**Review by acron0 — CHANGES_REQUESTED** [review]

File: src/witan/models/dem/ccm/core/projection_loop.clj (hunk 4 above)
Comment (seb231) [reply src/witan/models/dem/ccm/core/projection_loop.clj]: Should this be 1-0-0 then...?

File: src/witan/models/dem/ccm/core/projection_loop.clj (hunk 5 above)
Comment (seb231) [reply src/witan/models/dem/ccm/core/projection_loop.clj]: I wasn't sure if it made more sense just to get ride of them all at the same time?

File: src/witan/models/dem/ccm/core/projection_loop.clj (hunk 4 above)
Comment (acron0) [reply src/witan/models/dem/ccm/core/projection_loop.clj]: yep

File: src/witan/models/dem/ccm/core/projection_loop.clj (hunk 5 above)
Comment (acron0) [reply src/witan/models/dem/ccm/core/projection_loop.clj]: No pref

File: src/witan/models/dem/ccm/core/projection_loop.clj [hunk 6]
```diff
@@ -127,7 +127,7 @@
                             (ds/rename-columns {:popn-migrated :popn}))]
     {:latest-year-popn popn-w-migrants}))
 
-(defworkflowfn join-by-year
+(defworkflowfn join-by-year-1-0-0
```
Comment (acron0) [inline src/witan/models/dem/ccm/core/projection_loop.clj]: This needs to be changed in models.clj too

**Review by acron0 — CHANGES_REQUESTED** [review]

**Review by acron0 — APPROVED** [review]

**Pull request merged**